midend/local_copyprop.cpp

```diff
@@ -205,7 +205,16 @@ IR::AssignmentStatement *DoLocalCopyPropagation::preorder(IR::AssignmentStatemen
 }
 
 IR::AssignmentStatement *DoLocalCopyPropagation::postorder(IR::AssignmentStatement *as) {
-    if (as->left == as->right) {   // FIXME -- need deep equals here?
+    if (as->left->toString() == as->right->toString() &&
```

Comment:
For now, we could create a small `equiv` function that takes two `const IR::Expression *` and does some deeper comparisons (eg two PathExpressions with Paths with the same same) that would fix this specific case and not break other things, defined as a static function right above this.  Be sure to include a FIXME comment about this being temporary until a proper general deep comparison is implemented.